Civi/API/Service/CustomGroupService.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace Civi\API\Service;
+
+use Civi\Api4\CustomGroup;
+
+class CustomGroupService {
+  /**
+   * @var array
+   */
+  protected $cache = array();
+
+  /**
+   * @param $whereParts
+   *
+   * @return array|object
+   */
+  public function findBy($whereParts) {
```

Comment:
Ditto - this looks like a mini api for fetching data from the database.
What about using existing (cached) core functions instead?